### ISSUE TITLE
TextAlignment renders text in wrong location for External Fonts

### ISSUE THREAD

**Issue by Niaxor** [ForNeVeR/AvaloniaRider#454]
### Describe the bug

The Rider Previewer does not render open type fonts loaded as a static resource in the correct location in certain circumstances. The circumstance seems to depend on the text length, text alignment, and previewer zoom %.

For example, this center aligned text is rendering as right-aligned:

![386210660-5eeca490-0315-4cc8-8c04-433f8941ca0e](https://github.com/user-attachments/assets/a86a6aea-e661-45d1-a87e-a39a59efd938)

If I left-align it it appears as centered:

![386210928-5d1c25b5-0053-4a78-90b9-db3aa0c36f25](https://github.com/user-attachments/assets/1a0fa72b-6f74-42cc-885a-9d56311b8d48)

If I removed a single 't' from the end of the string it then renders in the correct location....?

I have tested with two fonts, both open types as static resource, both behaved inconsistently. I tested installed fonts like Arial and had no issues with placement. The placement of text is as expected during runtime, this is limited to the previewer.

### To Reproduce

- Define a open type font (Sinkin Sans e.g.) as a static resource in app.xaml
- Create a user control like this:
```
<UserControl xmlns="https://github.com/avaloniaui"
             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
             xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
             xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
             xmlns:reactiveUi="http://reactiveui.net"
             mc:Ignorable="d" d:DesignWidth="1920" d:DesignHeight="1000"
             Width="1920" Height="1000">
    <TextBlock VerticalAlignment="Center" HorizontalAlignment="Center" TextAlignment="Center" FontFamily="{StaticResource SinkinSans}" FontSize="70" Foreground="#AFC0CB" Text="Testttttt" ClipToBounds="False" />
</UserControl>
```
NOTE: If you don't set ClipToBounds="False" the text won't render at all since it is 'out of bounds'

-Set previewer to 50% (can test with various percentages)
-Play around with text length and see how the text behaves inconsistently

### Expected behavior

Preview TextBlocks as per runtime placement

Rider 2024.1.4
Plugin version: 1.3.2

**Comment by ForNeVeR** [comment]
Does the font render properly? Is it the right font? If so, then this is most likely the Avalonia bug and not of the plugin.

(Otherwise, it could be something related to how the resources are found when the plugin spins up the previewer, but I doubt that very much.)

A self-contained repro I can run would help.

**Comment by Niaxor** [comment]
The font renders properly, it is the correct font.

It just seems to be being rendered an additional half a screen to the right in some circumstances, only in the previewer. 

In the picture examples I provided above, if I set the text alignment to right, it will be off the screen all together (presumable placed at 1.5x the width of the screen). 

I will create a repro project for you :)

**Comment by Niaxor** [comment]
https://github.com/user-attachments/assets/392dafa6-b034-4ce7-9731-c4d548d08f66

Here is a demo showing that the previewer zoom setting is one thing that affects the placement. This hints to me that this is an issue with the previewer itself.

**Comment by Niaxor** [comment]
[fonttest.zip](https://github.com/user-attachments/files/17770383/fonttest.zip)

Repro project here.

To reproduce ->

0. Open in Rider
1. Build
2. Open UserControl1 Xaml 
3. Enable preview in Rider
4. Test all zoom percentages, and you should see the text position be incorrect for a number of them

**Comment by Niaxor** [comment]
Update: I made some tweaks to the repo and realised this IS actually affecting runtime, even though it is affected by zoom percentage of the previewer. 

I will re-open my first ticket on the Avalonia repo :'(

**Comment by Niaxor** [comment]
Turns out this was fixed in the latest release 11.2.1. It was indeed an issue with AvaloniaUI. Closing
